tests/e2e/testsuites/testsuites.go

```diff
@@ -544,15 +544,15 @@ func podLogs(client clientset.Interface, name, namespace string) ([]byte, error)
 }
 
 // Method to get cloudInstanceId from Node Labels
-func GetCloudInstanceIdFromNodeLabels(client clientset.Interface) (string, error) {
+func GetCloudInstanceIdFromNodeSpec(client clientset.Interface) (string, error) {
 	nodes, err := client.CoreV1().Nodes().List(context.TODO(), metav1.ListOptions{})
 	if err != nil {
 		return "", fmt.Errorf("error getting cloudInstanceId: %+v", err)
 	}
-
-	if cloudInstanceId, ok := nodes.Items[0].ObjectMeta.Labels[powervscloud.CloudInstanceIDLabel]; ok {
-		return cloudInstanceId, nil
+	nodeName := nodes.Items[0].Name
+	instanceInfo, err := powervscloud.GetInstanceInfoFromProviderID(client, nodeName)
```

Comment:
```suggestion
	instanceInfo, err := powervscloud.GetInstanceInfoFromProviderID(client, nodes.Items[0].Name)
```

Using index 0 directly is not a good idea, we need to explore how this can be addressed later in a different task.